nilearn/glm/first_level/first_level.py

```diff
@@ -124,26 +155,60 @@ def run_glm(Y, X, noise_model='ar1', bins=100, n_jobs=1, verbose=0):
     # Create the model
     ols_result = OLSModel(X).fit(Y)
 
-    if noise_model == 'ar1':
-        # compute and discretize the AR1 coefs
-        ar1 = (
-            (ols_result.residuals[1:]
-             * ols_result.residuals[:-1]).sum(axis=0)
-            / (ols_result.residuals ** 2).sum(axis=0)
-        )
+    if noise_model[:2] == 'ar':
+
+        err_msg = ('AR order must be a positive integer specified as arN, '
+                   'where N is an integer. E.g. ar3. '
+                   'You provied {0}.'.format(noise_model))
+        try:
+            ar_order = int(noise_model[2:])
+        except ValueError:
+            raise ValueError(err_msg)
+
+        # compute the AR coeficents
+        ar_coef_ = [_yule_walker(ols_result.residuals[:, res],
+                                 ar_order)
+                    for res in range(ols_result.residuals.shape[1])]
+        ar_coef_ = np.array(ar_coef_)
         del ols_result
-        ar1 = (ar1 * bins).astype(np.int) * 1. / bins
-        # Fit the AR model acccording to current AR(1) estimates
+        if len(ar_coef_[0]) == 1:
+            ar_coef_ = ar_coef_.ravel()
```

Comment:
```suggestion
            ar_coef_ = ar_coef_[:, 0]
```
less dangerous and more explicit

Reply:
Thanks, and I did a search for other ravels and didnt find any